File: daq/slc/src/dqm/src/Histo2S.cc
#include "Histo2S.h"

#include "ShortArray.h"

using namespace Belle2;

Histo2S::Histo2S() : Histo2()
{
  _data = new ShortArray();
}

Histo2S::Histo2S(const Histo2S& h) : Histo2(h)
{
  if (h._data != NULL) _data = new ShortArray(*h._data);
  else _data = new ShortArray();
}

Histo2S::Histo2S(const std::string& name, const std::string& title,
                 int nbinx, double xmin, double xmax,
                 int nbiny, double ymin, double ymax)
  : Histo2(name, title, nbinx, xmin, xmax, nbiny, ymin, ymax)
{
  _data = new ShortArray((nbinx + 2) * (nbiny + 2));
}

std::string Histo2S::getDataType() const throw()
{
  return std::string("H2S");
}

File: daq/slc/extra/javalib/css/org.csstudio.belle2.daq/src/org/belle2/daq/dqm/Histo2S.java
package org.belle2.daq.dqm;


public class Histo2S extends Histo2 {

	public Histo2S() {
		super();
		data = new ShortArray(0);
	}
	public Histo2S(String name, String title, int nbinx, double xmin, double xmax,
			int nbiny, double ymin, double ymax) {
		super(name, title, nbinx, xmin, xmax, nbiny, ymin, ymax);
		data = new ShortArray((nbinx+2)*(nbiny+2));
	}
	public Histo2S(Histo2S h) {
		super();
		data = new ShortArray(0);
	}
	public String getDataType() {
		return "H2S";
	}

}
